tests/SleepForDays/SleepForDaysWorkflowTests.cs

```diff
@@ -41,14 +42,33 @@ await worker.ExecuteAsync(async () =>
             var handle = await env.Client.StartWorkflowAsync(
                 (SleepForDaysWorkflow wf) => wf.RunAsync(),
                 new(id: $"wf-{Guid.NewGuid()}", taskQueue: worker.Options.TaskQueue!));
+
+            // Continuously check history to see if timer has started.
+            bool timerStarted = false;
+            while (!timerStarted)
+            {
+                var history = await handle.FetchHistoryAsync();
+                foreach (var e in history.Events)
+                {
+                    if (e.TimerStartedEventAttributes != null && (TimeSpan.FromDays(30) == e.TimerStartedEventAttributes.StartToFireTimeout.ToTimeSpan()))
+                    {
+                        timerStarted = true;
+                        break;
+                    }
+                }
+            }
+
+            // Sanity check - timer should have started.
+            Assert.True(timerStarted);
```

Comment:
The final assert can never be false. Would recommend only looping a certain number of times and sleeping a short bit between each fetch. 

Actually, I just realized even in samples we have a `AssertMore.EventuallyAsync` helper, so you could probably just have:

```csharp
await AssertMore.EventuallyAsync(async () =>
{
    var history = await handle.FetchHistoryAsync();
    Assert.Contains(history.Events, e => e.TimerStartedEventAttributes != null);
});
```

(untested but the idea is there)

Reply:
Oh much nicer. I've also updated the activity execution count assertions. By the time the timer event for the sleep/delay is in history, we should have already executed the activity once. Skipping 90 more days should then give us 4 total executions not 3, which was erroneous. I wonder why that passed, if the timer existed in history before, then we still would have had to execute the activity, we still should have had 4 executions not 3.